### PR TITLE
[TM-1144] reorder functions to get valid schema data for polygons

### PR DESCRIPTION
**Pull request by cesarLima1**
https://gfw.atlassian.net/browse/TM-1144

### CODE REVIEW

File: app/Services/PolygonService.php [hunk 1]
```diff
@@ -323,4 +347,59 @@ protected function transformAndStorePoints($geojson, $sitePolygonProperties): ar
...
+
+    public function isFieldInvalid($field, $value)
+    {
+        if (is_null($value) || $value === '') {
+            return true;
+        }
+
+        switch ($field) {
+            case 'plantstart':
+                return ! $this->isValidDate($value);
```
Comment (pachonjcl) [inline app/Services/PolygonService.php:359]: If you take a close look, I don't think you are following the same method naming.
you have `isSomethingValid` vs `isValidSomething`. 
My suggestion is to have `isInvalidField`, but do not mix

File: app/Services/PolygonService.php [hunk 2]
```diff
@@ -323,4 +347,59 @@ protected function transformAndStorePoints($geojson, $sitePolygonProperties): ar
...
+
+        return true;
+    }
+
+    private function isValidDate($date)
+    {
+        $d = DateTime::createFromFormat('Y-m-d', $date);
+
+        return $d && $d->format('Y-m-d') === $date;
+    }
```
Comment (pachonjcl) [inline app/Services/PolygonService.php:404]: isn't it possible that the DateTime parse throws an error? can you catch in the flow?

File: app/Services/PolygonService.php [hunk 3]
```diff
@@ -323,4 +347,59 @@ protected function transformAndStorePoints($geojson, $sitePolygonProperties): ar
...
+    {
+        $distributions = explode(',', $value);
+        foreach ($distributions as $distribution) {
+            if (! in_array(trim($distribution), self::VALID_DISTRIBUTIONS)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```
Comment (pachonjcl) [inline app/Services/PolygonService.php:397]: nitpick, but this two methods are exactly the same with two different parameters.

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: app/Services/PolygonService.php (hunk 1 above)
Comment (cesarLima1) [reply app/Services/PolygonService.php]: Done.

File: app/Services/PolygonService.php (hunk 2 above)
Comment (cesarLima1) [reply app/Services/PolygonService.php]: try catch added.

File: app/Services/PolygonService.php (hunk 3 above)
Comment (cesarLima1) [reply app/Services/PolygonService.php]: Using the same function for both now.

**Review by pachonjcl — APPROVED** [review]

**Pull request merged**